be/src/vec/functions/array/function_array_index.h

```diff
@@ -87,6 +96,58 @@ class FunctionArrayIndex : public IFunction {
 
     bool use_default_implementation_for_nulls() const override { return false; }
 
+    Status open(FunctionContext* context, FunctionContext::FunctionStateScope scope) override {
+        if (scope == FunctionContext::THREAD_LOCAL) {
+            return Status::OK();
```

Comment:
why？

Reply:
because this param in functionState is just read-only, which should be fragment-local